packages/babel-helper-define-polyfill-provider/src/imports-cache.ts

```diff
@@ -76,12 +82,12 @@ export default class ImportsCache {
       lastImport.parent === programPath.node &&
       lastImport.container === programPath.node.body
     ) {
-      lastImport = lastImport.insertAfter(node);
+      newNodes = lastImport.insertAfter(node);
     } else {
-      lastImport = programPath.unshiftContainer("body", node);
+      newNodes = programPath.unshiftContainer("body", node);
     }
-    lastImport = lastImport[lastImport.length - 1];
-    this._lastImports.set(programPath, lastImport);
+    const newNode = newNodes[newNodes.length - 1];
+    this._lastImports.set(programPath, newNode);
```

Comment:
modified this function [as requested](https://github.com/babel/babel-polyfills/pull/121#pullrequestreview-923386800), but this last line still has a type error (`Node` vs. `Program`)

Reply:
Actually, the error is correct and we should change the `_lastImport` type to `WeakMap<NodePath<t.Program>, NodePath<t.Node>>` (technically it's `WeakMap<NodePath<t.Program>, NodePath<t.ImportDeclaration | t.ImportStatement>>`, but it would require a bunch of type casts).

Reply:
`ImportStatement` didn't seem to be an available type. I tried `Import` and `ImportSpecifier` though, and they only errored on this same line, i.e. didn't introduce more errors as far as I could tell.

Changed it to `t.Node` for now though